cve_bin_tool/data_sources/osv_source.py

```diff
@@ -292,15 +292,18 @@ def format_data(self, all_cve_entries):
 
             severity_data.append(cve)
 
-            for package in cve_item["affected"]:
-                product = package["package"]["name"]
+            for package_data in cve_item.get("affected", []):
+                package = package_data.get("package", {})
+                if not package:
+                    continue
+
+                product = package.get("name")
                 vendor = (
-                    "unknown"  # OSV Schema does not provide vendor names for packages
-                )
-                if (
-                    "github.com/" in product
-                ):  # if package name is of format github.com/xxxx/yyyy  xxxx can be vendor name and yyyy is package name
-                    vendor = product.split("/")[-2]  # trying to guess vendor name
+                    "unknown"
+                )  # OSV Schema does not provide vendor names for packages
+
+                if "github.com/" in product:
```

Comment:
```suggestion
                if product.startswith("github.com/"):
```

This likely won't fix the codeQL complaint (which is irrelevant) but since we do technically only expect github.com to be at the start of the string, we might as well be more precise about this in case of weirdly formatted stuff.